mbed_devices/_internal/linux/device_detector.py

```diff
@@ -10,4 +14,27 @@ class LinuxDeviceDetector(DeviceDetector):
 
     def find_candidates(self) -> List[CandidateDevice]:
         """Return a list of CandidateDevices."""
-        pass
+        context = pyudev.Context()
+        return [
+            CandidateDevice(
+                mount_points=_find_fs_mounts_for_device(disk.properties.get("DEVNAME")),
+                product_id=disk.properties.get("ID_MODEL_ID"),
+                vendor_id=disk.properties.get("ID_VENDOR_ID"),
+                serial_number=disk.properties.get("ID_SERIAL_SHORT"),
+                serial_port=_find_serial_port_for_device(disk, pyudev.Context()),
+            )
+            for disk in context.list_devices(subsystem="block", ID_BUS="usb")
+        ]
+
+
+def _find_serial_port_for_device(disk: pyudev.Device, context: pyudev.Context) -> Any:
+    """Try to find a serial port associated with the given device."""
+    for tty_dev in context.list_devices(subsystem="tty"):
+        if tty_dev.properties.get("ID_SERIAL_SHORT") == disk.properties.get("ID_SERIAL_SHORT"):
```

Comment:
Could `disk.properties.get("ID_SERIAL_SHORT"):` be stored in a variable to avoid dictionary lookup on every loop?

Reply:
Good point. Yes, it could. Dictionary lookup is fast, and this is always a relatively small list, but variable lookup is definitely faster :).

Reply:
30126d2